### PR TITLE
[Embed form collections] Add documentation about twig macros

### PR DESCRIPTION
**Pull request by tcheymol**
Embeded for collections page.

Add documentation about how to use twig macros to fully customize the rendering of the forms generated with the prototype
The aim is to help people customizing their form rendering 

- Without having to write css in their javacsript
- Without adding css both to the js genrated forms and the twig genreated forms

### CODE REVIEW

**Comment by javiereguiluz** [comment]
@tcheymol thanks for your proposal.

Personally I've never seen something like this, so I'm asking our forms experts @yceruto and @xabbuh to see what's their opinion about adding this to the docs. Thanks.

**Comment by tcheymol** [comment]
@javiereguiluz @xabbuh @yceruto If I can be more specific on how I think it improoves the code : 

- The form layout will be the same for the forms generated with the prototype and the form generated on your twig template
- You can also add the bin icon to delete the form directly inside the macro
- Then it reduces the amount of javascript boilerplate needed to make the code work and improoves reusability.

I attached my codes sample if you want to see how much easier it is this way 

Template
Macro
Jquery

File: form/form_collections.rst [hunk 1]
```diff
@@ -720,3 +739,4 @@ the relationship between the removed ``Tag`` and ``Task`` object.
 .. _`JSFiddle`: http://jsfiddle.net/847Kf/4/
 .. _`@a2lix/symfony-collection`: https://github.com/a2lix/symfony-collection
 .. _`symfony-collection`: https://github.com/ninsuo/symfony-collection
+.. _`Twig macros`: https://twig.symfony.com/doc/2.x/tags/macro.html
```
Comment (noniagriconomie) [inline form/form_collections.rst]: Do twig doc provide a `current` doc? Defining a version like this is hard to maintain in the long run no?

File: form/form_collections.rst [hunk 2]
```diff
@@ -299,6 +299,25 @@ new "tag" forms. To render it, make the following change to your template:
 
         {{ form_widget(form.tags.vars.prototype.name)|e }}
 
+.. tip::
+
+    If you need to customize the rendering of your prototype, you can use `Twig macros`_ For instance, you can do :
+
+    .. code-block:: html+twig
+
+        {% macro displayTagForm(tag) %}
```
Comment (noniagriconomie) [inline form/form_collections.rst]: What about https://symfony.com/doc/current/form/form_themes.html#creating-a-form-theme-in-the-same-template-as-the-form ?

I do think that the other way is more a "best practice"

Comment (tcheymol) [reply form/form_collections.rst]: That is arguable, but I think you are right 😂. I found the macro way cool and concise, so maybee I did get a litle too much enthousiastic about it.
Closing then ? Thx

**Pull request closed without merging**